src/main/java/teammates/ui/webapi/action/GetSessionResultsAction.java

```diff
@@ -37,10 +38,17 @@ public void checkSpecificAccessControl() {
         Intent intent = Intent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
         switch (intent) {
         case INSTRUCTOR_RESULT:
-            InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.getId());
-            gateKeeper.verifyAccessible(instructor, fs);
+            if (userInfo == null) {
+                throw new UnauthorizedAccessException("Instructor account is required to access this resource");
+            } else {
+                InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+                gateKeeper.verifyAccessible(instructor, fs);
+            }
```

Comment:
I think here can be more consistent with style by just using if as guarding clause.